AST.fs

```diff
@@ -14,8 +14,20 @@ open Starling.Core.Var.Types
 /// </summary>
 [<AutoOpen>]
 module Types =
+    type SourcePosition = 
+        { StreamName: string; Line: int64; Column: int64; }
+        override this.ToString() = sprintf "SourcePosition { StreamName = \"%s\"; Line = %d; Column = %d; };" this.StreamName this.Line this.Column
+
+    /// A Node in the AST which annotates the data with information about position
+    //type Node<'a> = { lineno : int; Node : 'a; }
+    type Node<'a> = 
+        { Position: SourcePosition; Node: 'a }
+        static member (|>>) (n, f) = { Position = n.Position; Node = f n.Node }
+        static member (|=>) (n, b) = { Position = n.Position; Node = b }
+        override this.ToString() = sprintf "<%A: %A>" this.Position this.Node
+
     /// A Boolean operator.
-    type Bop =
+    type BopTypes =
```

Comment:
Still not sure about the name of this (sure, I'm being pedantic)—the plural here makes it sound like this is going to be a list or a product type to me.  Maybe something like `BopInner`?  idk.